src/FFmpegWriter.cpp

```diff
@@ -1222,6 +1222,13 @@ AVStream *FFmpegWriter::add_video_stream() {
 	st->avg_frame_rate = av_inv_q(c->time_base);
 	st->time_base.num = info.video_timebase.num;
 	st->time_base.den = info.video_timebase.den;
+#if (LIBAVFORMAT_VERSION_MAJOR >= 58)
+	_Pragma ("GCC diagnostic push");
+	_Pragma ("GCC diagnostic ignored \"-Wdeprecated-declarations\""); \
+	st->codec->time_base.num = info.video_timebase.num;
+	st->codec->time_base.den = info.video_timebase.den;
+	_Pragma ("GCC diagnostic pop");
```

Comment:
```suggestion
	#pragma GCC diagnostic push
	#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
	st->codec->time_base.num = info.video_timebase.num;
	st->codec->time_base.den = info.video_timebase.den;
	#pragma GCC diagnostic pop
```

The only reason we had to use the `_Pragma()` form in FFmpegUtilities is that you can't use preprocessor-macro `#pragma` _in_ a preprocessor macro. But it works fine here.